### ISSUE TITLE
[tune] turn metrics into an optional kwarg in PTL TuneReportCallback

### ISSUE THREAD

**Issue by sumanthratna** [ray-project/ray#11344]
### Describe your feature request
[from the docs](https://docs.ray.io/en/master/tune/tutorials/tune-pytorch-lightning.html#codecell5:~:text=This%20callback%20will%20take%20the%20avg_val_loss,as%20the%20loss%20and%20mean_accuracy%2C%20respectively.):
> ```python
> from ray.tune.integration.pytorch_lightning import TuneReportCallback
> callback = TuneReportCallback({
>     "loss": "avg_val_loss",
>     "mean_accuracy": "avg_val_accuracy"
> }, on="validation_end")
> ```
> This callback will take the `avg_val_loss` and `avg_val_accuracy` values from the PyTorch Lightning trainer and report them to Tune as the loss and mean_accuracy, respectively.

I think it makes sense to make `metrics` optional. If the metrics dict isn't passed, the callback should just report all of PTL's metrics to tune with the same names. 

---

CC @richardliaw 

edit: this functionality exists for XGBoost, so I think it makes sense to do this for PTL too

**Comment by richardliaw** [comment]
cc @krfricke ?

**Comment by krfricke** [comment]
Right, it exists for all other callbacks so we should add it here, too. I can do this soon, or would you like to take a pass on that @sumanthratna?

**Comment by sumanthratna** [comment]
@krfricke I'm pretty busy this week and next week so can you implement this one? Thanks!
